Updater: Add code documentation

Dentomologist opened the pull request:
Add a comment at the beginning of AutoUpdate.h giving an overview of
the update process and how the flow of control proceeds during an
update.

Rumi-Larry commented:
It would be better to add this as a separate readme file, and citing that file inside the cpp and h

Dentomologist commented:
Documentation moved to docs/autoupdate_overview.txt, with comments thereto in autoupdate related classes.

Rumi-Larry commented:
> 
> 
> Documentation moved to docs/autoupdate_overview.txt, with comments thereto in autoupdate related classes.

Good, is it necessary to cite the document in the mac updater files too?

leoetlino commented on docs/autoupdate_overview.txt:
```diff
@@ -0,0 +1,90 @@
+Dolphin's autoupdate procedure is spread through a number of files; this overview describes the
```
It might be worth converting this to Markdown? "General notes", "Class and file responsibilities", "AutoUpdateChecker" etc. would naturally be section headers.

Dentomologist commented:
- Converted text file to Markdown
- Added documentation pointer comments to MacUpdater/AppDelegate.mm, MacUpdater/main.m, and WinUpdater/Main.cpp
- Updated existing comments to reflect change from .txt to .md
- Fixed linter issue in MacUpdater/AppDelegate.mm
